File: projects/04/factorial.rb
#! /usr/bin/ruby
# factorial.rb displays a table of logarithms
#
# Begun by: Prof. Adams, for CS 214 at Calvin College.
# Completed by: Daniela Contreras ldc6
# Date: 02/07/22
# Input:  n can be any integer
# Precondition: n must be any number > 0
#    
# Output:factorial of n 

def factorial (n)
      answer = 1

      for i in 2..n do
         answer *= i
      
      end
      return answer
end

if __FILE__ == $0
   print "Enter n value: "
####below code -> GET THESE VALUES AS NUMBERS
   n = gets.chomp.to_f ###initializing variable Start
   print n
   print "! = "
   puts factorial(n)
  
end

File: projects/04/Factorial.java
/* LogTable.java displays a table of logarithms.
 *
 * Input: n as a double.
 * PRE: n is any number.
 * Output: A factorial number of n.
 * Date: 02/07/22
 * Begun by: Nelesen, for CS 214 at Calvin College.
 * Completed by: Daniela Contreras ldc6
 ********************************************************/
import java.util.Scanner; 		// Get input

class Factorial{

    public static double factorial(double n) {
      
      double answer = 1.0;

      for (int i = 2; i <= n; i++) {
        answer *= i;
      }

      return answer;
   }

    public static void main(String[] args){
    Scanner keyboard = new Scanner(System.in);
    /* System.out.println("To display n factorial,");
    System.out.println("enter n: ");*/
    System.out.println ("\nTo compute n! \n enter n: ");
    double n;
    n = keyboard.nextDouble();
    
    System.out.println ("\nFactorial(n) is:  " + factorial(n));
}
  }
    


   